### PR TITLE
Fixup waves adapter

### PR DESCRIPTION
**Pull request by cujomalainey**
Fixups for Waves adapter

### CODE REVIEW

**Review by btian1 — APPROVED** [review]

File: src/audio/module_adapter/module/waves.c [hunk 1]
```diff
@@ -232,31 +232,31 @@ static int waves_effect_check(struct comp_dev *dev)
 	/* todo use fallback to comp_verify_params when ready */
 
 	/* resampling not supported */
-	if (src_fmt->rate != audio_stream_get_rate(snk_fmt)) {
+	if (audio_stream_get_rate(src_fmt) != audio_stream_get_rate(snk_fmt)) {
```
Comment (lyakh) [inline src/audio/module_adapter/module/waves.c:235]: oh, yeah, that's the punishment for being too inventive with names, we couldn't possibly have guessed to grep for `src_fmt` as a buffer stream name ;-)

File: src/audio/module_adapter/module/waves.c [hunk 2]
```diff
@@ -232,31 +232,31 @@ static int waves_effect_check(struct comp_dev *dev)
...
+	if (audio_stream_get_frm_fmt(src_fmt) != audio_stream_get_frm_fmt(snk_fmt)) {
 		comp_err(dev, "waves_effect_check() source %d sink %d sample format mismatch",
 			 audio_stream_get_frm_fmt(src_fmt), audio_stream_get_frm_fmt(snk_fmt));
 		ret = -EINVAL;
 		goto out;
 	}
 
 	/* different interleaving is not supported */
-	if (source_c->buffer_fmt != sink_c->buffer_fmt) {
+	if (audio_stream_get_buffer_fmt(src_fmt) != audio_stream_get_buffer_fmt(snk_fmt)) {
```
Comment (lyakh) [inline src/audio/module_adapter/module/waves.c:259]: ok, these ones should've been caught, but looks like they were wrong too

**Review by lyakh — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]
wow, waves.c had a lot of missed places

**Comment by kv2019i** [comment]
Build failures are caused by a regression (to be fixed by https://github.com/thesofproject/sof/pull/7949 ), rest looks good as expected as this is only touching code not covered by CI.

**Comment by cujomalainey** [comment]
> Build failures are caused by a regression (to be fixed by https://github.com/thesofproject/sof/pull/7949 ), rest looks good as expected as this is only touching code not covered by CI.

Working to fix that so this is covered by CI going forward :)

**Pull request merged**